examples/econais/ec19d/sdk_example_kaa_client/src/kaa_client.c

```diff
@@ -0,0 +1,823 @@
+                                                            , kaa_client->kaa_context->channel_manager
+                                                            , kaa_client->kaa_context->bootstrap_manager);
+
+        if (error_code) {
+            KAA_LOG_ERROR(kaa_client->kaa_context->logger,
+                    error_code,
+                    "Failed to create log upload strategy");
+            return error_code;
+        }
+
+        error_code = ext_log_upload_strategy_by_volume_set_threshold_count(kaa_client->log_upload_strategy_context
+                                                                         , KAA_DEMO_UPLOAD_COUNT_THRESHOLD);
+        if (error_code) {
+            KAA_LOG_ERROR(kaa_client->kaa_context->logger,
+                    error_code,
+                    "Failed to create log upload strategy by volume set threshold count to %d",
+                    KAA_DEMO_UPLOAD_COUNT_THRESHOLD);
+            return error_code;
+        }
+
+        error_code = kaa_logging_init(kaa_client->kaa_context->log_collector
+                                    , kaa_client->log_storage_context
+                                    , kaa_client->log_upload_strategy_context);
+
+        if (error_code) {
+            KAA_LOG_ERROR(kaa_client->kaa_context->logger,
+                    error_code,
+                    "Failed to logging init");
+            return error_code;
+        }
+        KAA_LOG_INFO(kaa_client->kaa_context->logger,
+                KAA_ERR_NONE,
+                "Log collector init complete");
+        return error_code;
+
+}
+
+/*
+ * Example code for logging
+ */
+kaa_error_t kaa_client_log_record(kaa_client_t *kaa_client, const char *record)
+{
+    if (!kaa_client || !record) {
+        return KAA_ERR_BADPARAM;
+    }
+
+    kaa_logging_record_t * kaa_record = kaa_logging_record_create();
+    if (!kaa_record) {
+        KAA_LOG_ERROR(kaa_client->kaa_context->logger,
+                KAA_ERR_NOT_INITIALIZED,
+                "Failed to allocate log record");
+        return KAA_ERR_NOT_INITIALIZED;
+    }
+
+    kaa_record->body = kaa_string_move_create(record, NULL);
+
+    //Wait until thread sleep in select()
+    sndc_sem_wait(&kaa_client->logging_semophore);
+
+    kaa_error_t error_code = kaa_logging_add_record(kaa_client->kaa_context->log_collector, kaa_record);
+    if (error_code) {
+        KAA_LOG_ERROR(kaa_client->kaa_context->logger,
+                error_code,
+                "Failed to add log record");
+    }
+
+    KAA_LOG_DEBUG(kaa_client->kaa_context->logger,
+                    KAA_ERR_NONE,
+                    "Kaa record %s logged", record);
+
+    kaa_record->destroy(kaa_record);
+
+
+    return KAA_ERR_NONE;
+}
+
+/**
+ * Example code for configuration update
+ */
+kaa_error_t kaa_client_configuration_update(kaa_client_t *kaa_client, const kaa_root_configuration_t *configuration)
+{
+    if (!kaa_client || !configuration) {
```

Comment:
Please, use KAA_RETURN_IF_NIL2 where it is possible to do this, to be consistent with code style